### DISCUSSION TITLE
Can we get the prediction layer weights of ESM2 to predict the masked amino acids?

### DISCUSSION THREAD

**Discussion by sirius777coder** [facebookresearch/esm#306] [Q&A]
Hi all,

Thanks for the great project on ESM2 and ESMFold! I use the ESM2 to predict some masked amino acids for my specific tasks, and I think the self-supervised model ESM2 is a good choice. But I find there are no available linear prediction weights of ESM2 to predict the amino acid identity. Can we get this final layer of parameters? By the way, could I ask when will ESMFold release the code?

**Comment by tomsercu** [comment]
https://github.com/facebookresearch/esm/blob/main/esm/model/esm2.py#L129
Here the logits [batch x L x vocabsize] are computed from the final hidden layer; the logits are unnormalized log-probabilities (put them through softmax to get probabilities)
